## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [1]:
import pandas as pd
import numpy as np 

df = pd.read_csv('data.csv') 
data = np.array(df['data'])
print(data)

[17.49945292 19.66239904  7.18282343 29.84162485  9.23938649 14.94951581
 23.0172885  12.52827148 13.65228669 14.54595431 23.30872427 33.75324808
 20.24923636 13.29244733 23.23434992 16.4230418  19.88521702 27.05000732
 15.5127743  20.05415151 14.73135264 19.06139498 21.53942271 14.06732571
 17.9670682  18.58289581 16.17406993 12.87432628 11.47269664 19.07902883
 18.38565824 33.38820073  5.39139454 20.67635903 22.22266722 28.15780318
 23.01114324 14.93471778 20.00005857 23.25411543 18.11895082 24.62607043
  8.79145607 30.387108   28.80606806 17.98593597 23.66804468 20.28782355
 15.02518827 20.52626131 26.00219532 17.71344489 17.74598346 19.55317542
 22.60097798 27.67027538 16.19192417 23.05037746 21.29669604  8.84832568
 17.48410111 19.20602661 19.76257856 21.9560206   7.75806171 20.27753314
 15.93394654 11.36336584 23.14577858 24.41167746 16.08049839 25.05473769
 17.71090111 20.39893405 13.40756632 29.50692234  4.04330326 19.45128426
 24.17071763  7.79920072 18.86318441 19.53668801 24

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [2]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [4]:
from scipy import stats 

hypothesized_mean = 20 
t_stat, p_value = stats.ttest_1samp(data, hypothesized_mean) 

p_value_one_sided = p_value / 2
 
print(f"T-статистика: {t_stat}")
print(f"P-значення (одностороннє): {p_value_one_sided}")

if p_value_one_sided < 0.05 and t_stat > 0:
    print("Є статистично значуща різниця: середній чек підвищився.")
else:
    print("Немає статистично значущої різниці: середній чек не підвищився.")



T-статистика: -0.9953477389335053
P-значення (одностороннє): 0.1609968919505994
Немає статистично значущої різниці: середній чек не підвищився.


2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [5]:
df_2 = pd.read_csv('data_2.csv') 
data_2 = np.array(df['data'])
print(data_2)

[17.49945292 19.66239904  7.18282343 29.84162485  9.23938649 14.94951581
 23.0172885  12.52827148 13.65228669 14.54595431 23.30872427 33.75324808
 20.24923636 13.29244733 23.23434992 16.4230418  19.88521702 27.05000732
 15.5127743  20.05415151 14.73135264 19.06139498 21.53942271 14.06732571
 17.9670682  18.58289581 16.17406993 12.87432628 11.47269664 19.07902883
 18.38565824 33.38820073  5.39139454 20.67635903 22.22266722 28.15780318
 23.01114324 14.93471778 20.00005857 23.25411543 18.11895082 24.62607043
  8.79145607 30.387108   28.80606806 17.98593597 23.66804468 20.28782355
 15.02518827 20.52626131 26.00219532 17.71344489 17.74598346 19.55317542
 22.60097798 27.67027538 16.19192417 23.05037746 21.29669604  8.84832568
 17.48410111 19.20602661 19.76257856 21.9560206   7.75806171 20.27753314
 15.93394654 11.36336584 23.14577858 24.41167746 16.08049839 25.05473769
 17.71090111 20.39893405 13.40756632 29.50692234  4.04330326 19.45128426
 24.17071763  7.79920072 18.86318441 19.53668801 24

In [8]:
from statsmodels.stats.weightstats import ztest


mean_july = np.mean(data)
mean_august = np.mean(data_2)
std_july = np.std(data, ddof=1)
std_august = np.std(data_2, ddof=1)

n_july = len(data)
n_august = len(data_2)
pooled_se = np.sqrt(std_july**2/n_july + std_august**2/n_august)

z_stat = (mean_july - mean_august) / pooled_se 

p_value = 2 * (1 - stats.norm.cdf(abs(z_stat)))

mean_july, mean_august, z_stat, p_value

(np.float64(19.37755319670441),
 np.float64(19.37755319670441),
 np.float64(0.0),
 np.float64(1.0))

Як бачимо середній чек в липні та серпні був однаковим. (19.37755319670441)
Немає різниці між середніми показниками двох місяців (0.0)
Надзвичайно високе p-значення(1.0), яке свідчить про те, що немає статистично значущої різниці між середніми значеннями

Тобто не маємо підстав вважати, що середній чек за серпень статистично відрізняється від середнього чеку за липень 2022 року